src/libtiled/tilelayer.cpp

```diff
@@ -400,6 +410,7 @@ void TileLayer::rotate(RotateDirection direction)
     mWidth = newWidth;
     mHeight = newHeight;
     mChunks = newLayer->mChunks;
+    updateBounds();
```

Comment:
Why not `mBounds = newLayer->mBounds;`? (and of course, passing `mX` and `mY` to constructor when creating `newLayer`. This goes for all these functions creating a new layer.